Ensure ignored tests are picked up by tox

lnsndn opened the pull request:
This fixes #2757, see full description of motivation there.

The change modifies file permissions on `range_test.py` and `sge_test.py` from 755 to 644 so that they are actually picked up by tox. This revealed a bunch of failing tests that I also fixed. In some cases because the tests were wrong, in other cases because of actual problems in the luigi functionality. The commit messages contain detailed explanations of the changes.

The change has been tested by running the concerned files manually with `python -m unittest <module>`, and also by running the full tox test suite on my machine, verifying that 1) the tests are picked up, 2) they go through, 3) no other tests are impacted.

Would like @lallea to review the parts concerning MonthInstantiationTest.

lallea commented:
The changes to MonthInstantiationTest all look fine. Thx @lnsndn!

For context: The changed tests were passing before https://github.com/spotify/luigi/pull/2666 got merged, which corrected the confused RangeMonthly implementation. Before this PR, the tests validated the old, confused behaviour.

lnsndn commented:
Seems like the codecov/patch step fails because I changed some file handling operations. I noticed the way files were handled wasn't safe (they were opened but not closed) so I added functionality for closing them while I was making the tests work in the same code parts. Now codecov is complaining I didn't add a test for it. Is this vital to fix?

lnsndn commented:
Thank you @dlstadther. How do I get it merged?

dlstadther commented:
Thanks for your contribution @lnsndn ! (And thanks @lallea for your review and context!)
